File: conversions/markdown-table/elide-markdown-table.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-markdown-table

This demonstrates calling the TypeScript markdown-table from Python
using Elide's polyglot capabilities.

Benefits:
- One implementation shared across TypeScript and Python
- Consistent behavior across services
- Zero dependencies in both languages
- Perfect for Documentation, CLI tools
"""

print("=== Python Calling TypeScript markdown-table ===\n")

# NOTE: Exact syntax depends on Elide's Python polyglot API
# Assuming: from elide import require
#           markdown_table = require('./elide-markdown-table.ts')

# Example 1: Basic Usage
# def basic_example():
#     """Demonstrate basic functionality"""
#     # Use the TypeScript module from Python
#     result = markdown_table.default(input_data)
#     print(f"Result: {result}")
#     print()

# Example 2: Real-world Application
# class MarkdownTableService:
#     def __init__(self):
#         self.module = markdown_table
#     
#     def process(self, data):
#         """Process data using TypeScript implementation"""
#         return self.module.default(data)

# Example 3: API Integration
# class APIHandler:
#     def handle_request(self, request_data):
#         """Handle API request using shared logic"""
#         processed = markdown_table.default(request_data)
#         return {"status": "success", "data": processed}

# Example 4: Data Pipeline
# def data_pipeline(items):
#     """Process items in pipeline"""
#     results = []
#     for item in items:
#         result = markdown_table.default(item)
#         results.append(result)
#     return results

# Example 5: Batch Processing
# class BatchProcessor:
#     def __init__(self):
#         self.processed_count = 0
#     
#     def process_batch(self, batch):
#         """Process batch of items"""
#         for item in batch:
#             result = markdown_table.default(item)
#             self.processed_count += 1
#         return self.processed_count

print("Real-world use cases:")
for use_case in ['Documentation', 'CLI tools', 'Reports', 'GitHub PRs', 'Data visualization']:
    print(f"- {use_case}")
print()

print("Problem Solved:")
print("Before: Different implementations in TypeScript vs Python")
print("After: One Elide implementation = consistent behavior")
print()

print("Performance Benefits:")
print("- Zero cold start overhead with Elide")
print("- Instant execution")
print("- No dependencies needed")
print("- Shared runtime across languages")
print()

print("When Elide Python API is ready:")
print("  from elide import require")
print("  markdown_table = require('./elide-markdown-table.ts')")
print("  result = markdown_table.default(data)")

File: converted/utilities/markdown-table/elide-markdown-table.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-markdown-table
#
# This demonstrates calling the TypeScript markdown-table implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One markdown-table implementation shared across TypeScript and Ruby
# - Consistent generate markdown tables from data arrays across microservices
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript markdown-table ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# markdown_table_module = Elide.require('./elide-markdown-table.ts')

# Example: Basic usage
# result = markdown_table_module.default()
# puts "Result: #{result}"

puts "Real-world use case:"
puts "- Ruby worker uses generate markdown tables from data arrays"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent behavior across services"
puts "- Perfect for documentation generation, report formatting, data export"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────┐"
puts "│   Elide markdown-table (TypeScript)         │"
puts "│   conversions/markdown-table/elide-markdown-table.ts│"
puts "└─────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same generate markdown tables from data arrays everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby custom + JavaScript = different behaviors"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/markdown_table.rb"
puts "  MARKDOWN_TABLE = Elide.require('./elide-markdown-table.ts')"
